### ISSUE TITLE
Request textDocument/completion failed.

### ISSUE THREAD

**Issue by jcloutz** [HvyIndustries/crane#274]
Platform : darwin
VSCode Version : 1.10.2
Crane Version : 0.3.4
---

The following errors occured while writing mocks for using PHPUnit tests. I'm assuming that it has something to do with the this morning's update because it was working fine yesterday. Right not completion isn't working at all.

```
[Error - 8:54:15 AM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'label' of null
  Code: -32603 
[Error - 8:55:25 AM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'label' of null
  Code: -32603 
TypeError: Cannot read property 'kind' of undefined
    at parser.read_function_argument_list (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/function.js:188:21)
    at parser.recursive_variable_chain_scan (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/variable.js:107:29)
    at parser.read_variable (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/variable.js:69:17)
    at parser.read_expr_item (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/expr.js:344:19)
    at parser.read_expr (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/expr.js:12:21)
    at parser.read_argument_list (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/function.js:210:17)
    at parser.read_function_argument_list (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/function.js:186:29)
    at parser.recursive_variable_chain_scan (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/variable.js:107:29)
    at parser.read_variable (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/variable.js:69:17)
    at parser.read_expr_item (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/expr.js:344:19)
TypeError: Cannot read property 'kind' of undefined
    at parser.read_function_argument_list (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/function.js:188:21)
    at parser.recursive_variable_chain_scan (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/variable.js:107:29)
    at parser.read_variable (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/variable.js:69:17)
    at parser.read_expr_item (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/expr.js:344:19)
    at parser.read_expr (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/expr.js:12:21)
    at parser.read_argument_list (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/function.js:210:17)
    at parser.read_function_argument_list (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/function.js:186:29)
    at parser.recursive_variable_chain_scan (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/variable.js:107:29)
    at parser.read_variable (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/variable.js:69:17)
    at parser.read_expr_item (/Users/jeremycloutier/.vscode/extensions/HvyIndustries.crane-0.3.4/server/node_modules/php-parser/src/parser/expr.js:344:19)
[Error - 9:08:00 AM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'label' of null
  Code: -32603 
[Error - 9:08:23 AM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'label' of null
  Code: -32603 
[Error - 9:10:36 AM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'label' of null
  Code: -32603 
[Error - 9:10:44 AM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'label' of null
  Code: -32603 
[Error - 9:12:02 AM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'label' of null
  Code: -32603 
[Error - 9:12:28 AM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: Cannot read property 'label' of null
  Code: -32603 
```

**Comment by jcloutz** [comment]
A little more detail using the following code block

```php
$controller = $this->getMockBuilder($this->_getControllerClass())
    ->setMethods(array(
        '_callParent'
    ))
    ->getMock();

$filterExp = $this->getMockBuilder('App\IFilterExpression')->getMockForAbstractClass();

$controller->expects($this->once())
    ->method('_callParent') // after typing '->' the 'Request textDocument/completion failed.' error occurs
    ->with($this->_getControllerClass().'::foo')
    ->willReturn($filterExp);

$this->assertSame(
    $filterExp, // after typing the ',' the 'TypeError: Cannot read property 'kind' of undefined' error occurs'
    $this->_invokeReflectionMethod($controller, 'getFilterExpression', $raw)
);
```

**Comment by ichiriac** [comment]
Hi @jcloutz,

Thanks for the report.

That seems to fail on something like a variable based nested call, and fails into the arguments lists, something like `$foo->bar($foo->baz($arg`

Could you provide a little code/snippet of what you are writing in order to be able to reproduce the same error.

It may be a bug on an external library (php-parser with flag suppressErrors and a bad-formed/null AST node)

**Comment by ichiriac** [comment]
This bug is not new, but the latest release may be more verbose on errors. It will be fixed for the next 0.3.5 release

**Comment by ichiriac** [comment]
Stack for the label error : 

```
TypeError: Cannot read property 'label' of null
    at SuggestionBuilder.build (/home/devbox/Documents/hvy/crane-glayzzle/client/server/suggestionBuilder.js:205:23)
    at connection.onCompletion.e (/home/devbox/Documents/hvy/crane-glayzzle/client/server/server.js:79:42)
    at handleRequest (/home/devbox/Documents/hvy/crane-glayzzle/client/server/node_modules/vscode-jsonrpc/lib/main.js:237:37)
    at process.callback (/home/devbox/Documents/hvy/crane-glayzzle/client/server/node_modules/vscode-jsonrpc/lib/main.js:483:13)
```

**Comment by nevadascout** [comment]
Released in v0.3.5
